optosim/simulation/event_reader.py

```diff
@@ -71,172 +61,64 @@ def __init__(self, filenames):
 
         A.P. Colijn
         """
-        self.filenames = filenames
-        self.file_index = 0
-        self.file = None
-        self.event_names = []
-        self.event_index = -1
-
-        self.nfiles = len(self.filenames)
-        print("number of files: ", len(self.filenames))
-        self.read_config()
-
-    def read_config(self):
-        """Reads the configuration of the data reader
-
-        Parameters
-        ----------
-        None
-
-        Returns
-        -------
-        None
-
-        A.P. Colijn
-        """
-        if self.nfiles > 0:
-            with h5py.File(self.filenames[0], "r") as hf:
-                self.config = json.loads(hf.attrs["config"])
-
-            hf.close()
-
-    def print_config(self):
-        """Prints the configuration of the data reader
-
-        Parameters
-        ----------
-        None
 
-        Returns
-        -------
-        None
+        # Load config attribute from the first file
+        with h5py.File(filenames[0], 'r') as f:
+            self.config = json.loads(f.attrs.get("config"))
 
-        A.P. Colijn
-        """
-        print(self.config)
-
-    def __iter__(self):
-        """Returns an iterator for the data reader
-
-        Parameters
-        ----------
-        None
-
-        Returns
-        -------
-        self : EventReader
-            The iterator
-
-        A.P. Colijn
-        """
-        return self
-
-    def __next__(self):
-        """Reads the next event from the file
-
-        Parameters
-        ----------
-        None
-
-        Returns
-        -------
-        event_data : dict
-            The event data
-
-        A.P. Colijn
-        """
+        print("huh config = ", self.config)        
 
         if self.config['data_type_version'] == 1.0:
```

Comment:
This gives an error if the data_type_version is not in the config. As old data does not have it, I suggest updating with: 
`
        if self.config.get("data_type_version", 1.0) == 1.0:
`